File: segmentedControl/segmentedControl/ViewController.m
//
//  ViewController.m
//  segmentedControl
//
//  Created by Madala, Sarath Chandra on 9/10/18.
//  Copyright © 2018 Madala, Sarath Chandra. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *inputErr;
@property (weak, nonatomic) IBOutlet UILabel *btnErr;
@property (weak, nonatomic) IBOutlet UITextField *inputTextbox;
@property (weak, nonatomic) IBOutlet UITextField *outputTextbox;
- (IBAction)btnSort:(id)sender;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)segmentChange:(id)sender{
    //Segment actions goes here
    NSString *title = [segment titleForSegmentAtIndex:[segment selectedSegmentIndex]];
    [label setText:[NSString stringWithFormat:@"Enter %@s to sort", title]];
    
    _inputTextbox.text = @"";
    _outputTextbox.text = @"";
    
    if([segment selectedSegmentIndex] == 2){
        [_inputErr setHidden:FALSE];
        [_inputErr setText:[NSString stringWithFormat:@"Enter date in this format: YYYY/MM/DD"]];
    }
    else{
        [_inputErr setHidden:TRUE];
    }
}

- (IBAction)btnSort:(id)sender{
    
    //Take input string
    NSString *string = _inputTextbox.text;
    //Remove spaces
    string = [string stringByReplacingOccurrencesOfString:@" " withString:@""];
    NSLog(@"%@",string);
    
    if([segment selectedSegmentIndex] == 0){

        //String to array
        NSMutableArray<NSNumber *> *arr = [[string componentsSeparatedByString:@","] mutableCopy];
        NSLog(@"temp=%@",arr);
        
        int numLength = (int)[arr count];
        for (int i = 0; i < numLength-1; i++)
        {
            for(int j = i+1; j < numLength; j++)
            {
                NSNumber *temp=[arr objectAtIndex:i];
                int first =(int)[temp integerValue];
                
                NSNumber *temp1=[arr objectAtIndex:j];
                int second =(int)[temp1 integerValue];
                
                if(first > second)
                {
                    [arr replaceObjectAtIndex:j withObject:temp];
                    [arr replaceObjectAtIndex:i withObject:temp1];
                }
            }
        }
        
        NSString * result = [[arr valueForKey:@"description"] componentsJoinedByString:@", "];
        
        _outputTextbox.text = result;
    }
    else if([segment selectedSegmentIndex] == 1){
        
        //String to array
        NSMutableArray<NSString *> *arr = [[string componentsSeparatedByString:@","] mutableCopy];
        NSLog(@"temp=%@",arr);
        
        [arr sortUsingSelector:@selector(localizedCaseInsensitiveCompare:)];
        
        NSString * result = [[arr valueForKey:@"description"] componentsJoinedByString:@", "];
        
        _outputTextbox.text = result;
    }
    else{
        
        NSMutableArray *arr = [[string componentsSeparatedByString:@","] mutableCopy];
        NSLog(@"temp=%@",arr);
        
        /**/
        
        int numLength = (int)[arr count];
        for (int i = 0; i < numLength-1; i++)
        {
            for(int j = i+1; j < numLength; j++)
            {
                NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
                [formatter setDateFormat:@"yyyy/MM/dd"];
                
                NSString *temp=[arr objectAtIndex:i];
                NSDate *date = [formatter dateFromString:temp];
                
                NSString *temp1=[arr objectAtIndex:j];
                NSDate *date1 = [formatter dateFromString:temp1];
                
                
                
                
                if ([date compare:date1] == NSOrderedDescending) {
                    NSLog(@"%@ is later than %@", [arr objectAtIndex:i], [arr objectAtIndex:j]);
                    [arr replaceObjectAtIndex:j withObject:temp];
                    [arr replaceObjectAtIndex:i withObject:temp1];
                } else if ([date compare:date1] == NSOrderedAscending) {
                    NSLog(@"%@ is earlier than %@", [arr objectAtIndex:i], [arr objectAtIndex:j]);
                } else{
                    NSLog(@"dates are equal");
                }
            }
        }
        
        NSString * result = [[arr valueForKey:@"description"] componentsJoinedByString:@", "];
        
        _outputTextbox.text = result;
    }
}

- (BOOL)textFieldShouldReturn:(UITextField *)theTextField {
    if (theTextField == _inputTextbox) {
        [_inputTextbox resignFirstResponder];
    }
    return YES;
}

@end

File: Calculator/Caluculator/ViewController.swift
//
//  ViewController.swift
//  Caluculator
//
//  Created by Chava, Sravya on 11/20/18.
//  Copyright © 2018 Chava, Sravya. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    var numberOnScreen: Double = 0;
    var previousNumber: Double = 0;
    var performMath = true;
    var operation: Int = 0
    @IBOutlet weak var label: UILabel!
    @IBAction func numbers(_ sender: UIButton) {
        if performMath == true{
            if sender.tag == 19
            {
            label.text = "."
            numberOnScreen = Double(label.text!)!
            performMath = false
            }
            else
            {
            label.text = String(sender.tag-1)
            numberOnScreen = Double(label.text!)!
            performMath = false
            }
        }
        else
        {
            label.text = label.text! + String(sender.tag-1)
            print (label.text as Any)
            numberOnScreen = Double(label.text!)!
        }
    }
    @IBAction func buttons(_ sender: UIButton) {
        if label.text != "" && sender.tag != 11 && sender.tag != 16
        {
            previousNumber = Double(label.text!)!
            if sender.tag == 12
            {
               label.text = "/"
            }
            else if sender.tag == 13
            {
                label.text = "X"
            }
            else if sender.tag == 14
            {
                label.text = "-"
            }
            else if sender.tag == 15
            {
                    label.text = "+"
            }
            else if sender.tag == 17
            {
                label.text = "+/-"
            }
            else if sender.tag == 18
            {
                label.text = "%"
            }
            operation = sender.tag
            performMath = true
        }
        else if sender.tag == 16
        {
            if operation == 12
            {
                label.text = String(previousNumber / numberOnScreen)
            }
            else if operation == 13
            {
                label.text = String(previousNumber * numberOnScreen)
            }
            else if operation == 14
            {
                label.text = String(previousNumber - numberOnScreen)
            }
            else if operation == 15
            {
                label.text = String(previousNumber + numberOnScreen)
            }
            else if operation == 17
            {
                if previousNumber < 0
                {
                    label.text = "+" + String(previousNumber)
                }
                if previousNumber > 0
                {
                    label.text = "-" + String(previousNumber)
                }
            }
            else if operation == 18
            {
                label.text = String(previousNumber .truncatingRemainder(dividingBy: numberOnScreen))
            }
        }
        else if sender.tag == 11
        {
            label.text = ""
            previousNumber = 0
            numberOnScreen = 0
            operation = 0
        }
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }


}

